clcache.py

```diff
@@ -148,29 +148,28 @@ def cacheDirectory(self):
         return self.dir
 
     def clean(self, stats, maximumSize):
-        with self.lock:
-            currentSize = stats.currentCacheSize()
-            if currentSize < maximumSize:
-                return
+        currentSize = stats.currentCacheSize()
```

Comment:
`clean` is called under the cache lock (i've guarded each top-level instant of `stats` with the lock)